python/packages/autogen-ext/src/autogen_ext/models/openai/_openai_client.py

```diff
@@ -779,20 +779,45 @@ async def create_stream(
             # set the stop_reason for the usage chunk to the prior stop_reason
             stop_reason = choice.finish_reason if chunk.usage is None and stop_reason is None else stop_reason
             maybe_model = chunk.model
+
+            def process_reasoning(reasoning_deltas, content):
+                reasoning_deltas.append(content)
+                return "[Thought]" + content
+
             # First try get content
+            # Separate field acquisition reasoning process
+            if self._model_info["family"] == ModelFamily.R1:
+                if choice.delta.model_extra.get("reasoning_content"):
+                    if len(choice.delta.reasoning_content) > 0:
+                        yield process_reasoning(reasoning_deltas, choice.delta.reasoning_content)
```

Comment:
Can we use `is_reasoning` field in the following way for emitting chunks:

1. If `is_reasoning = False`, and the reasoning content is available in this chunk, emit a reasoning chunk prefixed by `<think>` if it is not already there. 
2. If `is_reasoning = True`, and the reasoning content is available in this chunk, emit a regular chunk, no prefix.
3. If `is_reasoning = True`, and the reasoning content isn't available in this chunk, set the `is_reasoning` to `False`, and prefix the chunk with `</think>` if the last chunk does not end with it already, emit this chunk. 

This way, the consumer of `create_stream` gets the same behavior as they would expect from local R1 models as remotely hosted one.